syncd/FlexCounter.cpp

```diff
@@ -1005,6 +1005,12 @@ class CounterContext : public BaseCounterContext
                         {
                             counter_data->setStatsMode(instance_stats_mode);
                         }
+                        auto it_vid = m_objectIdsMap.find(vid);
+                        if (it_vid != m_objectIdsMap.end())
+                        {
+                            // Remove and re-add if vid already exists
```

Comment:
this line is confusing, can you explain here ?

Reply:
When the ID's associated with the counter vid are updated, existing logic just does m_objectIdsMap.emplace(vids) which is a no-op. 

Thus, the fix is to remove the key vid from the m_objectIdsMap map and add a new entry with updated ID's